hw/xbox/xid.c

```diff
@@ -327,28 +336,35 @@ static void usb_xid_handle_control(USBDevice *dev, USBPacket *p,
     /* HID requests */
     case ClassInterfaceRequest | HID_GET_REPORT:
         DPRINTF("xid GET_REPORT 0x%x\n", value);
-        if (value == 0x100) { /* input */
-            assert(s->in_state.bLength <= length);
-            // s->in_state.bReportId++; /* FIXME: I'm not sure if bReportId is just a counter */
-            memcpy(data, &s->in_state, s->in_state.bLength);
-            p->actual_length = s->in_state.bLength;
+        if (value == 0x0100) { /* input */
+            if (length <= s->in_state.bLength) {
```

Comment:
Why do we allow `length` to be less than `s->in_state.bLength` here but below we stall until `length == s->out_state.length`?